platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -1007,6 +1007,8 @@ static void schedule_order_lock(unsigned lock_index)
 		       monitor32(&rctx->rwin->olock[lock_index],
 				 __ATOMIC_ACQUIRE) != rctx->sn)
 			doze();
+	rctx->rwin->lock_index = lock_index;
```

Comment:
The reorder window (rwin) is shared by all threads, you can't store per-thread data in it, you would use the reorder context (rctx) for that.

But why are we saving the lock_index? Is it used for making any correctness checks (e.g. verifying that the application isn't unlocking some lock it hasn't acquired)?

Reply:
As noted above, since `odp_schedule_order_unlock_lock()` only takes one parameter (the index of the new lock to be acquired), we need a place to store the index of the lock that the caller now holds to be used as an implied parameter to this API. So this is supposed to be a thread-local variable. If that's not what's happening here then that needs to be fixed, so thanks for pointing that out.

Reply:
The initial draft of ordered lock considered API for locking and unlocking ordered locks within a context. We found an use-case from the customer where the application was splitting a single ordered flow to two different flows, which is possible with our current API architecture but the drawback is that it requires the application to make two different calls odp_schedule_unlock() and odp_schedule_lock() which will have performance issues since the implementation has to internally switch out of the lock and then acquire the new lock. The main objective of this new API is to optimize this use-case by using a single API which switches the ordered context from one ordered lock to another.

Reply:
I will be moving lock_index from rctx to sched_scalable_thread_state_t in the next version and thanks for pointing it out.